lib/charms/mongodb_libs/v0/relations.py

```diff
@@ -0,0 +1,171 @@
+                return
+
+            database_dbs = mongo.get_databases()
+            relation_dbs = self._get_databases_from_relations(departed_relation_id)
+            for database in database_dbs - relation_dbs:
+                logger.info("Drop database: %s", database)
+                mongo.drop_database(database)
+
+    def _get_config(self, username: str) -> MongoDBConfiguration:
+        """Construct config object for future user creation."""
+        relation = self._get_relation(username)
+        return MongoDBConfiguration(
+            replset=self.charm.app.name,
+            database=self._get_relation_database(relation),
+            username=username,
+            password=generate_password(),
+            hosts=self.charm.mongodb_config.hosts,
+            roles=self._get_relation_roles(relation),
+        )
+
+    def _set_relation(self, config: MongoDBConfiguration):
+        """Save all output fields into application relation."""
+        relation = self._get_relation(config.username)
+        if relation is None:
+            return None
+        relation.data[self.charm.app]["username"] = config.username
+        relation.data[self.charm.app]["password"] = config.password
+        relation.data[self.charm.app]["database"] = config.password
+        relation.data[self.charm.app]["endpoints"] = ",".join(config.hosts)
+        relation.data[self.charm.app]["replset"] = config.replset
+        relation.data[self.charm.app]["uris"] = config.uri
+
+    @staticmethod
+    def _get_username(relation_id: str) -> str:
+        """Construct username."""
+        return f"relation-{relation_id}"
+
+    def _get_users_from_relations(self, departed_relation_id: Optional[int]):
+        """Return usernames for all relations."""
+        relations = self.model.relations[REL_NAME]
+        return set([
+            self._get_username(relation.id)
+            for relation in relations
+            if relation.id != departed_relation_id
+        ])
+
+    def _get_databases_from_relations(self, departed_relation_id: Optional[int]) -> Set[str]:
+        """Return database names from all relations."""
+        relations = self.model.relations[REL_NAME]
+        return set([
+            self._get_relation_database(relation)
+            for relation in relations
+            if relation.id != departed_relation_id
+        ])
+
+    def _get_relation(self, username: str) -> Relation:
+        """Parse relation ID from a username and return Relation object."""
+        match = re.match(r"^relation-(\d+)$", username)
+        # We generated username in `_get_users_from_relations`
+        # func and passed it into this function later.
+        # It means the username here MUST match to regex.
+        assert match is not None, "No relation match"
+        relation_id = int(match.group(1))
+        return self.model.get_relation(REL_NAME, relation_id)
+
+    def _get_relation_database(self, relation: Relation) -> str:
```

Comment:
```suggestion
    def _get_database_from_relation(self, relation: Relation) -> str:
```
follow naming scheme

Reply:
fixed in https://github.com/delgod/mongodb-k8s-operator/pull/3/commits/847a2ed26d6e9b11a2606b414e108ed8650f447d